jsonnet/components/control-plane.libsonnet

```diff
@@ -46,12 +47,74 @@ function(params)
       },
     },
 
+    operationalServiceMonitorEtcd: self.serviceMonitorEtcd {
```

Comment:
we can probably factor out the generation of a new service monitor from the "full" version?

```
operationalServiceMonitorEtcd: serviceMonitorForScrapeProfile(
  self.serviceMonitorEtcd, // the original service monitor definition
  "operational", // the resulting scrape profile
  ['etcd_disk_backend_commit_duration_seconds_bucket', ...], // list of metrics being kept.
)
```

Reply:
Do you mean abstracting the logic to create a jsonnet function? I'm just not sure because some ServiceMonitors have multiple endpoints, that export different metrics, see kubelet for an example. So what would the interface of that function look like? 

While working on https://issues.redhat.com/browse/MON-2694 I thought we could also just simply add the list of metrics to all the endpoints, this would allow for something like what you suggested.

However, the direction I was taking, [MON-2694](https://issues.redhat.com/browse/MON-2694) was more aligned with the goal of adding for each endpoint only the relevant subset of metrics. The tool would either fetch these metrics from a pre-compiled list or from a file that Openshift developers could provide as a parameter, both the list and the file would have a structure like so:
- namespace/serviceMonitorName/port/path:  []metrics

WDYT? @simonpasquier

Reply:
> Do you mean abstracting the logic to create a jsonnet function?

Yes

> I'm just not sure because some ServiceMonitors have multiple endpoints, that export different metrics, see kubelet for an example. So what would the interface of that function look like?

Good point. Thinking more about it, the "operational" profile is all about keeping a set of "known" metrics so the same list could be applied to all endpoints?

Also the current kubelet monitor endpoints have drop configurations in place that aren't "useful" for the operational profile since it knows which metrics to keep and dropping the drop  (sic) directives would be a bit more efficient.

E.g. the existing kubelet service monitor:

```
spec:
  endpoints:
  - bearerTokenFile: ""
    honorLabels: true
    interval: 30s
    metricRelabelings:
    - action: drop
      regex: kubelet_(pod_worker_latency_microseconds|pod_start_latency_microseconds|cgroup_manager_latency_microseconds|pod_worker_start_latency_microseconds|pleg_relist_latency_microseconds|pleg_relist_interval_microseconds|runtime_operations|runtime_operations_latency_microseconds|runtime_operations_errors|eviction_stats_age_microseconds|device_plugin_registration_count|device_plugin_alloc_latency_microseconds|network_plugin_operations_latency_microseconds)
      sourceLabels:
      - __name__
  ...
  - bearerTokenFile: ""
    honorLabels: true
    honorTimestamps: false
    interval: 30s
    metricRelabelings:
    - action: drop
      regex: container_(network_tcp_usage_total|network_udp_usage_total|tasks_state|cpu_load_average_10s)
      sourceLabels:
      - __name__
...
```

could be transformed into an operational service monitor looking like:

```
spec:
  endpoints:
  - bearerTokenFile: ""
    honorLabels: true
    interval: 30s
    metricRelabelings:
    - action: keep
      regex: "<list of operational metrics>"
      sourceLabels:
      - __name__
    port: https-metrics
...
    path: /metrics
  - bearerTokenFile: ""
    honorLabels: true
    honorTimestamps: false
    interval: 30s
    metricRelabelings:
    - action: keep
      regex: "<list of operational metrics>"
      sourceLabels:
      - __name__
    path: /metrics/cadvisor
...
```

Reply:
Got it, that makes sense. I just have one more question: 
Should we generate our ServiceMonitors with jsonnet and have that in this PR or should we redact that part from this PR and include it in the PR where we will provide [MON-2694](https://issues.redhat.com/browse/MON-2694)? I like more this idea because that way we would be using our own solution for generating the ServiceMonitors instead of using jsonnet? The tool would just consume the assets folder and output yaml files to the same folder. But let me know if I'm missing something, maybe instead of doing things with a dedicated tool we could just create jsonnet package that other developers could use 

WDYT @simonpasquier, if you want we can also quickly discuss it on a call